api/git/types/types.go

```diff
@@ -0,0 +1,19 @@
+package gittypes
+
+// RepoConfig represents a configuration for a repo
+type RepoConfig struct {
+	// The repo url
+	URL string `example:"https://github.com/portainer/portainer-ee.git"`
+	// The reference name
+	ReferenceName string `example:"refs/heads/branch_name"`
+	// Path to where the config file is in this url/refName
+	ConfigFilePath string `example:"docker-compose.yml"`
+	// Git credentials
+	Authentication *GitAuthentication
+	ConfigHash     []byte
+}
+
+type GitAuthentication struct {
+	Username string
+	Password string
+}
```

Comment:
@DimaSalakhov Hi Dmitry, I see those fields are not  included in EE-161, should we remove them? cos I see you are using `ConfigHash` in your commits

Reply:
@ArrisLee 
Not sure about EE-161, maybe they aren't needed in there, I wouldn't expect EE-161 to have all the necessary changes we'd need for EE-248.
In EE-248, on the other hand, we do need to store `Username` and `Password` to be able to pull a new version of a private repo. `ConfigHash` is used to store the hash of all config files, so that we can compare file hashes instead of plain-text file contents when we try to figure out if files had changed.

Reply:
yep, I think so too, so the struct here is ok right?

Reply:
Thoughts:
- I believe we don't use `ConfigFilePath` field anywhere
- would prefer to rename `RepoConfig` to `GitConfig`, at least because we also have `GitAuthentication`
- would prefer not to have a subpackage for types, especially with a single `types.go` file, it's not javascript :)